docs-src/extensionFunctions.adoc

```diff
@@ -0,0 +1,20 @@
+=== Extension functions
+
+Extension functions allow us to attach custom functions to any library or custom code we write. You can extend String, Collection or 
+any other known type with some custom behavior. Given example creates `isContainedBy` extension function that we can directly call on 
+any kind of `T` type, as though it was part of the standard library.
```

Comment:
Some thoughts... Maybe site could be similar in terms of colors to this color space (for brand consistency), but that's up to you, guys :)

![image](https://user-images.githubusercontent.com/6358061/27759607-072ba5ba-5e35-11e7-818f-0c25cc2013b6.png)

Did we go markdown for easier contribution?

I think we could have some basic / advanced examples distinction, so that novice users coming to the website could go to basic area, and more experienced ones to *advanced*. 

Anyways - these are just thoughts. I'll open new PR with this basic example.

Cheers,
*/root*